tests/umock_c_2_int/CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+#Copyright (c) Microsoft. All rights reserved.
+#Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#Note: this project only exists for the purpose of examining the result of precompilation of MOCKABLE_FUNCTION(s) in different contexts.
+
+#Note2: vld and producing correct code as a result of precompilation is not really possible with our current set.
+#       -Duse_vld:BOOL=ON on the cmake command line will result in /FIvld.h (Force Include vld.h)
+#       - so on top of the preprocessed file, vld.h will the (force) included resulting in all the symbols that vld.h brings to be doubly defined.
+#       - this results in a non-compilable file.
+#       - therefore this project is only produced when use_vld is OFF.
```

Comment:
Hmmm, so, we will have a problem with all unittests and vld? Or am I reading this wrong?
If we do, I think we have to address the vld part (maybe by controlling VLD inclusion with a #define.

/FI is nice for not having to write #include "vld.h", buuuut if that means we lose the UT+vld combo, then we have to bite the bullet and include vld properly with an #include.

Reply:
You're reading it right.

By the way, `#include "vld.h"` can be done in the main.c/main.cpp, (and only once), so it is not a big change.

Let me write a task for this so we don't forget about it.

Reply:
[https://msazure.visualstudio.com/One/_workitems/edit/29596519](https://msazure.visualstudio.com/One/_workitems/edit/29596519)